tests/bunit.web.tests/TestDoubles/NavigationManager/FakeNavigationManagerTest.cs

```diff
@@ -146,4 +146,81 @@ public void Test009()
 
 		locationChangedInvoked.ShouldBeFalse();
 	}
+
+#if NET7_0_OR_GREATER
+	[Fact(DisplayName = "When component provides NavigationLock, FakeNavigationManager should intercept calls")]
+	public void Test010()
+	{
+		var fakeNavigationManager = CreateFakeNavigationManager();
+		var cut = RenderComponent<InterceptNavigateToCounterComponent>();
+
+		cut.Find("button").Click();
+
+		cut.Instance.NavigationIntercepted.ShouldBeTrue();
+		cut.WaitForAssertion(() => fakeNavigationManager.History.ShouldBeEmpty());
+	}
+
+	[Fact(DisplayName = "Intercepting external url's should work")]
+	public void Test011()
+	{
+		var fakeNavigationManager = CreateFakeNavigationManager();
+		var cut = RenderComponent<GotoExternalResourceComponent>();
+
+		cut.Find("button").Click();
+
+		fakeNavigationManager.History.ShouldNotBeEmpty();
+	}
+
+	private class InterceptNavigateToCounterComponent : ComponentBase
+	{
+		protected override void BuildRenderTree(RenderTreeBuilder builder)
+		{
+			builder.OpenElement(0, "button");
+			builder.AddAttribute(1, "onclick", EventCallback.Factory.Create<MouseEventArgs>(this,
+				() => NavigationManager.NavigateTo("/counter")
+			));
+			builder.AddContent(2, "Goto counter");
+			builder.CloseElement();
+			builder.AddMarkupContent(3, "\n\n");
+			builder.OpenComponent<NavigationLock>(4);
+			builder.AddAttribute(5, "OnBeforeInternalNavigation", Microsoft.AspNetCore.Components.CompilerServices.RuntimeHelpers.TypeCheck(EventCallback.Factory.Create<LocationChangingContext>(this,
+					InterceptNavigation
+				)));
+			builder.CloseComponent();
+		}
+
+		public bool NavigationIntercepted;
```

Comment:
```suggestion
		public bool NavigationIntercepted { get; set; }
```

This should either be lower cased and a field or keep the current case and be a property.